src/components/app/index.js

```diff
@@ -25,7 +24,7 @@ class App extends React.Component {
 
   componentDidUpdate(prevProps) {
     if (prevProps.data.schema_id !== this.props.data.schema_id) {
-      this.resetStoreData(formatData(this.props.data));
+      this.resetStoreData(loadData(this.props.data));
```

Comment:
Can you explain the change here i.e. difference between formatData and loadData?

Reply:
In this case, `loadData` just calls `formatData`, so it does the same thing. But it meant I could remove an extra dependency, and make the usage more consistent

Reply:
Does load data actually request data or is it more mapping data to a structure?

Reply:
(Maybe the naming could be changed what do you think?)

Reply:
`loadData` determines how data should be loaded (i.e. async from JSON, or imported from a JS file, or randomly-generated, or directly via props), then formats it and returns it.

Maybe `getData` (or similar) would be a better name, idk?

https://github.com/quantumblacklabs/kedro-viz/blob/develop/src/components/app/load-data.js

Reply:
Ah ok - reading the method loadData it seems like it's doing different things based on the parameter being passed to the method (sometimes passing a string / others an array). It sometimes loads from a mock data file, other times will load from a specific path, and lastly will format the data array if you supply it that structure.

To me it seems like this method is doing a lot and isn't very specific in its use, have you considered breaking this down a bit or is it more convenience? (Just something to consider rather than a blocker on this PR)

Reply:
Yeah I get that, but it's kind of necessary to have this switch statement to determine what data source to use dependent on the argument it gets passed, as this is the way that the application chooses a data source, and while most of them are synchronous, one is async. I don't think I'd change anything but perhaps the name, though it doesn't bother me hugely.

Reply:
Think it's more breaking the single responsibility idea but happy to take outside this PR 👍